doc/tutorials/chapter_8/02_finite_state_machine.md

```diff
@@ -0,0 +1,353 @@
+# Content
+
+- [Microwave Oven in Finite State Machine (FSM)](#microwave-oven-in-finite-state-machine-fsm)
+- [State](#state)
+- [Transitions](#transitions)
+- [Microwave Oven State Diagram](#microwave-oven-state-diagram)
+- [ROHD FSM](#rohd-fsm)
+- [FSM Simulation](#fsm-simulation)
+- [Exercise](#exercise)
+
+## Learning Outcome
+
+In this chapter:
+
+- You will learn how to create a microwave oven finite state machine using ROHD abstraction API.
+
+## Microwave Oven in Finite State Machine (FSM)
+
+Today, we going to develop an Microwave Oven FSM using ROHD. Since its just a case study, let make this microwave oven only consists of 4 states (standby, cooking, paused and completed).
+
+### State
+
+- Standby: LED Blue. The initial state, where the oven is idle and waiting for user input
+- Cooking: LED Yellow. The state in which the oven is actively cooking the food
+- Paused: LED Red. The state in which cooking is temporarily suspended, but can be resumed.
+- Completed: LED Green. The state in which the oven has finished cooking the food.
+
+### Transitions
+
+Transitions between states would be triggered by events such as button presses or sensor readings. For example, when the "start" button is pressed, the FSM would transition from "standby" to "cooking." When the cooking time has expired, it would transition from "cooking" to "completed." If the "pause" button is pressed while in the "cooking" state, the FSM would transition to "paused." And if the "resume" button is pressed while in the "paused" state, the FSM would transition back to "cooking."
+
+## Microwave Oven State Diagram
+
+![Oven FSM](./assets/oven_fsm.png)
+
+## ROHD FSM
+
+In ROHD, there are abstraction level of writting FSM. Yes, you can definitely wrote the code using Sequential and Combinational like previous chapter. But, today we want to see how we can leverage the abstraction layer provided in ROHD to quickly create the Oven FSM above.
+
+First, we want to import the ROHD package and also `counter` module. We can use the counter interface we created last session.
+
+```dart
+import 'package:rohd/rohd.dart';
+import './counter_interface.dart';
+```
+
+We also want to represent the standby, cooking, paused, and completed states and the buttons start, pause and resume in dart enums. However, we want to one-hot encoded the buttons with custom value with dart enhanced enums.
```

Comment:
one-hot would be `001`, `010`, `100`, not 0, 1, 2.  Also, what benefit do you get by one-hot encoding or using the dart special enums?  as you've numbered them, they are the same as `index`, and also those `value`s won't get used as the actual FSM state values in the hardware